csharp/src/Google.Protobuf.Test/CodedInputStreamAsyncTest.cs

```diff
@@ -0,0 +1,502 @@
+#region Copyright notice and license
```

Comment:
How would you suggest doing that? I suspect that trying to reduce the duplication will increase the complexity overall. I'm all for it if you can think of a good way of doing it.